medusa/clients/torrent/qbittorrent.py

```diff
@@ -36,30 +36,73 @@ def __init__(self, host=None, username=None, password=None):
     @property
     def api(self):
         """Get API version."""
+        # Update the auth method to v2
+        self._get_auth = self._get_auth_v2
+        # Attempt to get API v2 version first
+        self.url = '{host}api/v2/app/webapiVersion'.format(host=self.host)
+        try:
+            version = self.session.get(self.url, verify=app.TORRENT_VERIFY_CERT,
+                                       cookies=self.session.cookies)
+            # Make sure version is using the (major, minor, release) format
+            version = map(int, version.content.split('.'))
+            if len(version) < 2:
+                version.append(0)
+            return tuple(version)
+        except (AttributeError, ValueError) as error:
+            log.error('{name}: Unable to get API version. Error: {error!r}',
+                      {'name': self.name, 'error': error})
+
+        # Fall back to API v1
+        self._get_auth = self._get_auth_legacy
         try:
             self.url = '{host}version/api'.format(host=self.host)
             version = int(self.session.get(self.url, verify=app.TORRENT_VERIFY_CERT).content)
+            # Convert old API versioning to new versioning (major, minor, release)
+            version = (1, version % 100, 0)
         except Exception:
-            version = 1
+            version = (1, 0, 0)
         return version
 
     def _get_auth(self):
+        """This method gets overriden by the `api` property."""
+        return self._get_auth_v2() or self._get_auth_legacy()
 
-        if self.api > 1:
-            self.url = '{host}login'.format(host=self.host)
-            data = {
-                'username': self.username,
-                'password': self.password,
-            }
-            try:
-                self.response = self.session.post(self.url, data=data)
-            except Exception:
-                return None
+    def _get_auth_v2(self):
+        """Authenticate using the new method (API v2)."""
+        self.url = '{host}api/v2/auth/login'.format(host=self.host)
+        data = {
+            'username': self.username,
+            'password': self.password,
+        }
+        try:
+            self.response = self.session.post(self.url, data=data)
+        except Exception:
+            return None
 
-        else:
+        if self.response.status_code == 404:
+            return None
+
+        self.session.cookies = self.response.cookies
+        self.auth = self.response.content
+
+        return self.auth if not self.response.status_code == 404 else None
```

Comment:
The check is not needed
```suggestion
        return self.auth
```